math/pythagoras.cpp

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright 2020 @author beqakd
+ * @file
+ *
+ * A basic implementation of 3d distance between two point.
+ */
+
+#include <cassert>   // For testing(assert)
+#include <cmath>     // For std::pow(), std::sqrt() and so on
+#include <iostream>  // For io operations
+
+/**
+ * The distance between two points in a three dimensional - 3D - coordinate
+ * system. For this we have 3 point x, y, z.
+ * Also we created method int distance which takes points and returns distance.
+ */
+class point {
+ private:
+    int x;
+    int y;
+    int z;
+
+ public:
+    point(int a, int b, int c)
+        : x(a), y(b), z(c) {}  // Our constructor to fill coordinates.
+    int distance(const point &,
+                const point &);  // Our prototype of distance method.
+};
+
+/**
+ * The distance between two points in a three dimensional - 3D - coordinate
+ * system.
+ * @param point_fir 3D coordinates of first point.
+ * @param point_sec 3d coordinates of second point.
+ * @return returns the distance between this two Point.
+ */
+int point::distance(const point &a, const point &b) {
+    // calculate distance and return it.
+    int x_2 = std::pow(std::abs(b.x - a.x), 2);
+    int y_2 = std::pow(std::abs(b.y - a.y), 2);
+    int z_2 = std::pow(std::abs(b.z - a.z), 2);
+    return std::sqrt(x_2 + y_2 + z_2);
```

Comment:
Wouldn’t the sort be non-integer most likely? Return type double would be appropriate